app/routes/_gcn.circulars/circulars.server.ts

```diff
@@ -261,3 +291,198 @@ export async function circularRedirect(query: string) {
     throw redirect(circularURL)
   }
 }
+
+/**
+ * Gets a specific version of a given circular
+ *
+ * Throws an HTTP error if:
+ *  - The provided circularId isNaN
+ *  - The provided version isNaN
+ *  - No version is found matching the provided circularId and version
+ *
+ * @param circularId
+ * @param version
+ * @returns a Circular corresponding to the specific version and Id
+ */
+export async function getSpecificCircularVersion(
+  circularId: number,
+  version: number
+): Promise<Circular> {
+  if (isNaN(circularId) || isNaN(version))
+    throw new Response(null, { status: 404 })
+  const db = await tables()
+  const result = await db.circulars_history.get({
+    circularId,
+    version,
+  })
+  if (!result)
+    throw new Response(null, {
+      status: 404,
+    })
+  return result
+}
+
+/**
+ * Gets all entries in circulars_history for a given circularId
+ * @param circularId
+ * @returns an array of previous versions of a Circular sorted by version
+ */
+export async function getCircularHistory(
+  circularId: number
+): Promise<Circular[]> {
+  const db = await tables()
+  const result = await db.circulars_history.query({
+    KeyConditionExpression: 'circularId = :circularId',
+    ExpressionAttributeValues: {
+      ':circularId': circularId,
+    },
+  })
+  return result.Items as Circular[]
+}
+
+/**
+ * Creates a set of changes in circulars_change_requests for users
+ * who do not have moderator permissions
+ *
+ * Throws an HTTP error if:
+ *  - The subject is invalid
+ *  - The body is invalid
+ *  - The user is not signed in
+ *
+ * @param circularId
+ * @param body
+ * @param subject
+ * @param request
+ */
+export async function createChangeRequest(
+  circularId: number,
+  body: string,
+  subject: string,
+  request: Request
+) {
+  if (!subjectIsValid(subject))
+    throw new Response('subject is invalid', { status: 400 })
+  if (!bodyIsValid(body)) throw new Response('body is invalid', { status: 400 })
+
+  const user = await getUser(request)
+  if (!user)
+    throw new Response('User is not signed in', {
+      status: 403,
+    })
+  const requestor = formatAuthor(user)
+  const db = await tables()
+  await db.circulars_change_requests.put({
+    circularId,
+    body,
+    subject,
+    requestorSub: user.sub,
+    requestor,
+  })
+}
+
+/**
+ * Gets all change requests for a given circular
+ * @param circularId
+ * @returns
+ */
+export async function getChangeRequests(
+  circularId: number
+): Promise<CircularChangeRequest[]> {
+  const db = await tables()
+  return (
+    await db.circulars_change_requests.query({
+      KeyConditionExpression: 'circularId = :circularId',
+      ExpressionAttributeValues: {
+        ':circularId': circularId,
+      },
+    })
+  ).Items
+}
```

Comment:
Again, do we want this to return the entire content of all of the change requests, or just a list of them?